### DISCUSSION TITLE
K_points confusion

### DISCUSSION THREAD

**Discussion by achyuttt** [dftbplus/dftbplus#1437] [General]
Hello everyone, 
I am a new user to the DFTB+ and I am learning to make calculations for 2D COFs. But I am confused on the K points that I have to use for my calculations. How should I get those K points.

For examples I followed the recipes for TiO2 and calculated the Density of States but the K points used in the Band structure calculations

KPointsAndWeights = Klines {
    1   0.5   0.5  -0.5    # Z
   20   0.0   0.0   0.0    # G
   45   0.0   0.0   0.5    # X
   10   0.25  0.25  0.25   # P
  }

How are this points selected? 
Do anyone have idea regarding this?

Also, when I calculated DOS for my COFs, I got the same DOS for 7-8 trials but the DOS I got from DFT and DFTB are completely different. Currently I am doing my calculation for an unitcell of 84 atoms.

I would be happy to any discussions and answer to this question.

Thank you!

**Comment by aradi** [comment]
K-points for band structure plots are typically selected according to your Brillouin-zone. Look up the high symmetry k-points of the BZ of your lattice type.

**Reply by achyuttt** [reply]
Can you please explain how does the supercellfolding work? in the KpointsandWeight function of DFTB+? 

I have been trying to perform DFTB+ calculation for 2D Covalent Organic Frameworks since a week following the documentation of DFTB+ and its recipes available. But everytime, I am getting the same results for Density of states. What could be the reason?

Also, how is the geo.gen file generated in the recipe of graphene? I was using Burai-a software for getting atomic positions and lattice vectors but I am not sure if the coordinates I get are for primitive unit cell or not.


Can anyone please help me to generate the geometry.gen file for any material I want to do DFTB calculation and also how to write the K points for density of states calculation?

**Reply by aradi** [reply]
For DOS calculation, you typically use a dense Monkhorst pack mesh:
```KPointsAndWeights = SupercellFolding {
  n1  0  0
  0   n2  0
  0    0  n3
 0.5 0.5 0.5
}
```
with n1, n2, n3 being integer (even) numbers.

As for the geometry file: You can also generate VASP POSCAR files and read them in with 
```
Geometry = VaspFormat {
...
}
```
The POSCAR format is recognized by many tools, so you can check your geometries before doing the calculation.

**Reply by achyuttt** [reply]
Here what does those 0.5 signify?

How does 0.5 change sampling of the Brillouin zone?

For example for a 2D system, should n1 and n2 only be mentioned? and n3 =1 means 0.5 0.5 and 0 should be entered in the SupercellFolding?
